### DISCUSSION TITLE
Field Type Support

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [appwrite/appwrite#2909] [Ideas]
Hi 
Is there any plan to support following type of fileds?

- json
- jsonb
- text
- uuid
- date
- time
- timestamp
- geoLocation

Based on going through several documents and community comments i get to know date /time stamp should be stored in time integer to filtering details this some times should be more consious also since time() in php differs from js one. So a common way of approch like UTC timestamp as in mongodb might be more better.

**Comment by eldadfux** [comment]
Yes, we do plan to add more attribute types over time. GEO location queries are also in plan.
Can you elaborate more on the differences between PHP vs JS times and how this been a challenge for you?

**Reply by iamdineshkumar** [reply]
@eldadfux  For an example In PHP Time() function provide output 1647797270 in JS new Date().getTime() provide 1647797270304 which has difference. To handle it in PHP we need to do Time() value multiply by 1000 to convert it equalent to JS time. Hope this will give clarification.

**Reply by malikdoksoz** [reply]
> GEO location queries are also in plan.

Hello, thank you for your work. But it's been a year, is there any progress on this?

**Comment by stnguyen90** [comment]
FYI, [Version 1.0.0](https://github.com/appwrite/appwrite/releases/tag/1.0.0) now has DateTime types which is an ISO-8601 timestamp with millisecond precision.

**Comment by jayjah** [comment]
I just want to say that `JSON` and `JSONB` would be great to have!
